core/state-machine/src/proving_backend.rs

```diff
@@ -53,6 +52,36 @@ impl<'a, S, H> ProvingBackendEssence<'a, S, H>
 			.map(|x| x.map(|val| val.to_vec()))
 			.map_err(map_e)
 	}
+
+	pub fn record_all_keys(&mut self) {
+		let mut read_overlay = MemoryDB::default();
+		let eph = Ephemeral::new(
+			self.backend.backend_storage(),
+			&mut read_overlay,
+		);
+
+		let mut iter = move || -> Result<(), Box<TrieError<H::Out>>> {
+			let root = self.backend.root();
+			let trie = TrieDB::<H>::new(&eph, root)?;
+			let iter = trie.iter()?;
+
+			for x in iter {
+				let (key, _) = x?;
+
+				// there's currently no API like iter_with()
+				// => use iter to enumerate all keys AND lookup each
+				// key using get_with
+				trie.get_with(&key, &mut *self.proof_recorder)
+					.map(|x| x.map(|val| val.to_vec()))?;
```

Comment:
Why do we need this `map`? It looks like the resulting vec is immediately dropped?

Reply:
Yep, thanks. Could you please remove it in your PR? :)